File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/games/blackjack/CardPathKeys.java
package com.panopset.marin.oldswpkg.games.blackjack;

import com.panopset.blackjackEngine.Card;
import com.panopset.marin.fx.ImageMapCache;
import javafx.scene.image.Image;

public class CardPathKeys {

  private final String basePath;
  private final String ext;
  
  public CardPathKeys(String basePath, String ext) {
    this.basePath = basePath;
    this.ext = ext;
  }

  public Image getImage(Card card) {
    return find(card);
  }

  Image find(Card card) {
    Image rtn = ImageMapCache.get(card.name());
    if (rtn == null) {
      rtn = ImageMapCache.get(card.name(), getPathFor(card));
    }
    return rtn;
  }

  Image find(String id) {
    Image rtn = ImageMapCache.get(id);
    if (rtn == null) {
      rtn = ImageMapCache.get(id, getPathFor(id));
    }
    return rtn;
  }

  String getPathFor(Card card) {
    return getPathFor(getCardSvgName(card));
  }

  String getPathFor(String id) {
    return String.format("%s%s.%s", basePath, id, ext);
  }

  public String getCardSvgName(Card card) {
    return card.getCardDefinition().getKey();
  }

  public Image getBackBlue() {
    return find("1B");
  }

  public Image getBackRed() {
    return find("2B");
  }

}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/games/blackjack/CardPathKeys.kt
package com.panopset.marin.oldswpkg.games.blackjack

import com.panopset.blackjackEngine.Card
import com.panopset.marin.fx.ImageMapCache
import javafx.scene.image.Image

class CardPathKeys(private val basePath: String, private val ext: String) {
    fun getImage(card: Card): Image? {
        return find(card)
    }

    fun find(card: Card): Image? {
        var rtn = ImageMapCache[card.name()]
        if (rtn == null) {
            rtn = ImageMapCache[card.name(), getPathFor(card)]
        }
        return rtn
    }

    fun find(id: String?): Image? {
        var rtn = ImageMapCache[id!!]
        if (rtn == null) {
            rtn = ImageMapCache[id, getPathFor(id)]
        }
        return rtn
    }

    fun getPathFor(card: Card): String {
        return getPathFor(getCardSvgName(card))
    }

    fun getPathFor(id: String?): String {
        return String.format("%s%s.%s", basePath, id, ext)
    }

    fun getCardSvgName(card: Card): String {
        return card.cardDefinition.key
    }

    val backBlue: Image?
        get() = find("1B")
    val backRed: Image?
        get() = find("2B")
}
